src/query/loader.rs

```diff
@@ -0,0 +1,335 @@
+use sea_orm::{sea_query::ValueTuple, Condition, ModelTrait, QueryFilter};
+use std::{collections::HashMap, hash::Hash, marker::PhantomData, sync::Arc};
+
+use crate::apply_order;
+
+#[derive(Clone)]
+pub struct KeyComplex<T>
+where
+    T: sea_orm::EntityTrait,
+{
+    /// The key tuple to equal with columns
+    pub key: Vec<sea_orm::Value>,
+    /// Meta Information
+    pub meta: HashAbleGroupKey<T>,
+}
+
+impl<T> PartialEq for KeyComplex<T>
+where
+    T: sea_orm::EntityTrait,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.key.eq(&other.key) && self.meta.eq(&other.meta)
+    }
+}
+
+impl<T> Eq for KeyComplex<T> where T: sea_orm::EntityTrait {}
+
+impl<T> Hash for KeyComplex<T>
+where
+    T: sea_orm::EntityTrait,
+{
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.key.hash(state);
+        self.meta.hash(state);
+    }
+}
+
+#[derive(Clone)]
+pub struct HashAbleGroupKey<T>
```

Comment:
Nit: I think usually `HashAble` is one word?

Reply:
Renamed